generic_scripts/nlk.py

```diff
@@ -0,0 +1,167 @@
+import numpy as _np
+import matplotlib.pyplot as _plt
+from mathphys.constants import vacuum_permeability
+from mathphys.beam_optics import beam_rigidity
+
+
+class Wire:
+    """."""
+    def __init__(self, position, current):
+        self.pos = position  # Column vector
+        self.curr = current
+
+    def Bfield(self, x, y):
+        mu0 = vacuum_permeability
+        r_w = self.pos  # Wire positions
+        r = _np.array([x, y])[:, None]  # Target position
```

Comment:
it would be better to change the input of this method. you should accept as input the vector(s) where the field must be calculated, not the `x` and `y` positions separately.